src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_CallerInfoAttributes.cs

```diff
@@ -388,9 +387,8 @@ public static class C
 <default-arg-expression>");
         }
 
-        // PROTOTYPE(caller-expr): Should this have a warning?
         [ConditionalFact(typeof(CoreClrOnly))]
-        public void TestArgumentExpressionIsReferingToItself()
+        public void TestArgumentExpressionIsSelfReferential()
```

Comment:
>TestArgumentExpressionIsSelfReferential [](http://example.com/codeflow?start=20&length=39)

Do we have a test covering consumption of similar attribute from metadata? Perhaps we can compile similar class to image reference, confirm the attribute is there and test the same scenario against that image.